Encryption.md

```diff
@@ -187,18 +187,17 @@ Only the `FileCryptoMetaData` is written as a plaintext, all other file parts ar
 ### Plaintext footer mode
 
 This mode allows older Parquet versions (released before the encryption support) to access unencrypted 
-columns in encrypted files - at a price of exposing certain metadata fields in these files. It can be 
-useful during a transitional period in organizations 
+columns in encrypted files - at a price of leaving certain metadata fields unprotected in these files 
+(not encrypted or tamper-proofed). The plaintext footer mode can be useful during a transitional period 
```

Comment:
The 'Plaintext footer mode' could also be used in the scenario that no footer key with appropriate access control privilege being found. For example, in column access control scenario, the footer needs to have less or equal control than(to) any column because accessing any column requires accessing footer first.  If there are plaintext columns, the footer will need to be kept as plaintext.

Reply:
While this scenario is possible, the second part is not true. If there are plaintext columns, the footer can, and should be encrypted if you want to make the data protection stronger. If the readers access both encrypted and plaintext columns - it means they are already connected to a KMS and authentication system, and have no problem to get one more key, for the file footer. If readers work only with plaintext columns and don't want to authenticate / connect a KMS - you still can encrypt the footer with a simple key - for example derived from the file name itself, via a company-wide algorithm. This will be done locally in the reader, no authenticating or KMS access are required. While this is not as strong as working with KMS - its still better than not encrypting footer at all, and allows to protect against breaches from outside. The external attackers don't know the algorithm used in your company for footer key derivation. Again, this is better that plaintext footer, which is immediately visible to attackers, who can learn about file contents (schema/column names, crypto metadata, etc). But its ultimately up to you.

Reply:
Yeah, it is better than nothing. But deriving from filename could cause some problem. What if the filename is changed? But you are right, the footer can be encrypted separately by using some way that doesn't need KMS.

Reply:
I agree, file name change is a problem - but there is a neat solution to this. We have the footer key metadata field (in FileCryptoMetaData). One option is to keep the file name (or its hash) there, so its never lost. Another option is just to create a random string / byte array, and put it in this key metadata field. In any case, the footer key is calculated locally as F(key_metadata), where F is your function/algorithm.